nextpnr.dockerfile

```diff
@@ -106,6 +106,33 @@ RUN cd /tmp/nextpnr/build \
  && make -j $(nproc) \
  && make DESTDIR=/opt/nextpnr install
 
+
+#---
+
+FROM build AS build-gowin
+RUN mkdir -p /tmp/nextpnr/build \
+ && apt-get update -qq \
+ && DEBIAN_FRONTEND=noninteractive apt-get -y install --no-install-recommends python3-setuptools python3-pip  \
```

Comment:
I need some background about apicula/apycula. When building nextpnr-ice40 and nextpnr-ecp5, some assets are required, but not executables/scripts. If that is the case with nextpnr-gowin, then you should get the assets here instead of installing the Python package.

Apart from that, `icestorm` and `prjtrellis` are built in separated dockerfiles/workflows, from sources. Overall, each tool is built/installed once only. We should do the same for apicula/apycula. Then, this stage would COPY from `hdlc/pkg:apicula`.

Reply:
Actually is a python library which also provides a link for running it as a executable. The nextpnr-gowin needs the library itself so it would just save some symbolic links.

I will make the adjusts to be at pkg and then just copy :D

Reply:
> Actually is a python library which also provides a link for running it as a executable. The nextpnr-gowin needs the library itself so it would just save some symbolic links.

Does nextpnr-gowin actually call apicula/apycula Python scripts at runtime? I would expect it to be required when building nextpnr-gowin, but then it should be independent (as nextpnr-ice40 or nextpnr-ecp5).

> I will make the adjusts to be at pkg and then just copy :D

Thanks! You can take Symbiyosys' dockerfile as an example. However, that one has an explicit `make install`, which makes it easier to handle regardless of PIP. I had a quick look at apicula and I didn't find a similar feature.

Note that, when creating the package, the PREFIX of the tools needs to be `/usr/local`; and DESTDIR should be used for placing them in a temporary location (to be then copied to the package image). I don't know whether the PREFIX is relevant for apicula.

Reply:
Just checked, after building it it doesnt need it. So I removed from the nextpnr container. Apicula is python so there is no make install, just pip and easy_install which doesnt support prefix, only target which does not generate the "binaries".